src/tests/FastBoard_unittest.cpp

```diff
@@ -0,0 +1,342 @@
+    FastBoard b;
+    b.reset_board(5);  
+    b.set_square(2, 2, FastBoard::WHITE);
+    EXPECT_EQ(false, b.is_suicide(b.get_vertex(1, 1), FastBoard::BLACK));
+    EXPECT_EQ(false, b.is_suicide(b.get_vertex(2, 1), FastBoard::BLACK));
+}
+
+// Results will make more sense in FullBuard test
+TEST(FastBoardTest, IsSuicideWhenForBlackInAllWhiteField) {
+    FastBoard b = create_5x5_all_white_field();
+
+    EXPECT_EQ(false, b.is_suicide(b.get_vertex(1, 1), FastBoard::BLACK));
+    EXPECT_EQ(false, b.is_suicide(b.get_vertex(3, 3), FastBoard::BLACK));
+    EXPECT_EQ(false, b.is_suicide(b.get_vertex(4, 4), FastBoard::BLACK));
+    EXPECT_EQ(false, b.is_suicide(b.get_vertex(4, 2), FastBoard::BLACK));
+    EXPECT_EQ(false, b.is_suicide(b.get_vertex(4, 4), FastBoard::BLACK));
+}
+
+TEST(FastBoardTest, CalcAreaScore) {
+    FastBoard b = create_filled_5x5();
+    EXPECT_EQ(-6.5, b.area_score(6.5F)); 
+    EXPECT_EQ(-.5, b.area_score(0.5F));
+    EXPECT_EQ(-9.0, b.area_score(9.0F)); 
+}
+
+TEST(FastBoardTest, CalcAreaScoreOnWhiteField) {
+    FastBoard b = create_5x5_all_white_field();
+    EXPECT_EQ(-31.5, b.area_score(6.5F)); 
+    EXPECT_EQ(-25.5, b.area_score(0.5F));
+    EXPECT_EQ(-34.0, b.area_score(9.0F)); 
+}
+
+TEST(FastBoardTest, CalcAreaScoreOnSemiFilled9x9) {
+    FastBoard b = create_filled_9x9();
+    EXPECT_EQ(-7.5, b.area_score(6.5F)); 
+    EXPECT_EQ(-1.5, b.area_score(0.5F));
+    EXPECT_EQ(-10.0, b.area_score(9.0F)); 
+}
+
+TEST(FastBoardTest, ToMove) {
+    FastBoard b = create_filled_5x5();
+    EXPECT_EQ(FastBoard::BLACK, b.get_to_move());
+    EXPECT_EQ(true, b.black_to_move());
+    b.set_to_move(FastBoard::WHITE);
+    EXPECT_EQ(FastBoard::WHITE, b.get_to_move());
+    EXPECT_EQ(false, b.black_to_move());
+}
+
+TEST(FastBoardTest, MoveToText) {
+    FastBoard b = create_filled_3x3();
+    EXPECT_EQ("B1", b.move_to_text(b.get_vertex(1, 0)));
+    EXPECT_EQ("A2", b.move_to_text(b.get_vertex(0, 1)));
+    EXPECT_EQ("pass", b.move_to_text(FastBoard::PASS));
+    EXPECT_EQ("resign", b.move_to_text(FastBoard::RESIGN));
+}
+
+TEST(FastBoardTest, MoveToTextSgf) {
+    FastBoard b = create_filled_3x3();
+    EXPECT_EQ("bc", b.move_to_text_sgf(b.get_vertex(1, 0)));
+    EXPECT_EQ("ab", b.move_to_text_sgf(b.get_vertex(0, 1)));
+    EXPECT_EQ("ca", b.move_to_text_sgf(b.get_vertex(2, 2)));
+    EXPECT_EQ("tt", b.move_to_text_sgf(FastBoard::PASS));
+    EXPECT_EQ("tt", b.move_to_text_sgf(FastBoard::RESIGN));
+}
+
+TEST(FastBoardTest, GetStoneList) {
+    FastBoard emptyBoard;
+    emptyBoard.reset_board(3);
+    EXPECT_EQ("", emptyBoard.get_stone_list());
+    
+    FastBoard b = create_filled_5x5();
+    EXPECT_EQ("A4 B2 C2 C3 C4 C5 D2 D3", b.get_stone_list());
+    
+    FastBoard whiteFieldBoard = create_5x5_all_white_field();
+    EXPECT_EQ("A3 B3 C1 C2 C3 C4 C5 D3 D5 E4", whiteFieldBoard.get_stone_list());
+}
+
+TEST(FastBoardTest, StarPoint9x9) {
+    
+    EXPECT_EQ(true, FastBoard::starpoint(9, 2, 2)); 
```

Comment:
This test probably makes more sense if one uses SGF or GTP text coordinates rather than the numbers.

i.e. check that on 19x19 "D4" is indeed a star point, rather than the (2,2) numbers which make little sense outside Leela.

Reply:
If there were a text_to_move method, or if starpoint had a signature that accepted a string like "D4", then I would agree with you. But since there is not, this is difficult to do. I wouldn't want to add an extra method to FastBoard that is used only for testing purposes, and I don't want to add logic to the unit test code to do the conversion because then it would need testing too. Is it OK to leave this as is, or can you think of a nice way to do what you suggest?